### PR TITLE
Added SpanFromWorkflowContext function

### LINKED ISSUE #1701: go.temporal.io/sdk/contrib/datadog to export TracerSpan and allow custom SpanContextKey
**Is your feature request related to a problem? Please describe.**
In order to be able to access spans from `workflow.Context` to be able to do things like add custom tags, it would be helpful if the `go.temporal.io/sdk/contrib/datadog/tracing` package exported `TracerSpan` and allowed setting a custom `SpanContextKey` in the tracer options.

This would enable code like the following to be written:
```go
// Assuming the following import:
// import 	"go.temporal.io/sdk/contrib/datadog/tracing"

val := ctx.Value(DD_ACTIVE_SPAN_CONTEXT_KEY)
if val == nil {
	return
}

span := *val.(*tracing.TracerSpan)
span.SetTag("myCustomTag", myCustomValue)
```

**Describe the solution you'd like**
go.temporal.io/sdk/contrib/datadog to export the TracerSpan struct (currently named `tracerSpan` so it isn't exported) and allow custom SpanContextKey in the `TracerOptions` struct.

I have prepared some changes that solve this for my particular use-case which you can see here. Happy to open a PR if you agree that this feature makes sense for the library:
https://github.com/temporalio/sdk-go/compare/master...cdimitroulas:temporal-sdk-go:christos/export-datadog-tracer-span-struct?expand=1

**Describe alternatives you've considered**
I was not able to find an alternative solution that supports this use-case.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by cdimitroulas**
## What was changed
Added a `SpanFromWorkflowContext` function to allow access to the DataDog span object from within a workflow.

## Why?
This allows users to access the Span from within a workflow to (for example) add custom tags on the span.

## Checklist

1. Closes #1701 
2. How was this tested: New unit test added

3. Any docs updates needed? Would be nice, please let me know where I could add these if this is desired.

### CODE REVIEW

File: contrib/datadog/tracing/interceptor.go [hunk 1]
```diff
@@ -143,6 +143,18 @@ func (t *tracerImpl) ContextWithSpan(ctx context.Context, span interceptor.Trace
 	return tracer.ContextWithSpan(ctx, span.(*tracerSpan).Span)
 }
 
+// SpanFromWorkflowContext extracts the DataDog Span object from the workflow context.
+func SpanFromWorkflowContext(ctx workflow.Context) (ddtrace.Span, bool) {
```
Comment (cretz) [inline contrib/datadog/tracing/interceptor.go:147]: Not sure the bool is needed. We can just use `nil` to represent not present.

Comment (cdimitroulas) [reply contrib/datadog/tracing/interceptor.go]: Agreed - I've updated it now.

**Review by cretz — APPROVED** [review]
LGTM, will let @Quinn-With-Two-Ns review

File: contrib/datadog/tracing/interceptor.go [hunk 2]
```diff
@@ -143,6 +143,18 @@ func (t *tracerImpl) ContextWithSpan(ctx context.Context, span interceptor.Trace
 	return tracer.ContextWithSpan(ctx, span.(*tracerSpan).Span)
 }
 
+// SpanFromWorkflowContext extracts the DataDog Span object from the workflow context.
+func SpanFromWorkflowContext(ctx workflow.Context) ddtrace.Span {
```
Comment (Quinn-With-Two-Ns) [inline contrib/datadog/tracing/interceptor.go:147]: We should match the equivalent [SpanFromContext](https://github.com/DataDog/dd-trace-go/blob/58e1e634e199676dd2ec5ce89600da91a282b253/ddtrace/tracer/context.go#L25) signature here, so return a ` (Span, bool)` and return a noop span if none is set

**Review by Quinn-With-Two-Ns — CHANGES_REQUESTED** [review]

File: contrib/datadog/tracing/interceptor.go (hunk 2 above)
Comment (cdimitroulas) [reply contrib/datadog/tracing/interceptor.go:147]: @Quinn-With-Two-Ns how can I created a noop span? The `NoopSpan` is exported from `gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal` so it's not accessible outside the datadog tracing package

Comment (cdimitroulas) [reply contrib/datadog/tracing/interceptor.go:147]: We could implement our own NoopSpan here but this seems like a lot of extra code for little benefit

File: contrib/datadog/tracing/interceptor.go (hunk 1 above)
Comment (cretz) [reply contrib/datadog/tracing/interceptor.go]: Actually, I agree with @Quinn-With-Two-Ns below concerning noop span

File: contrib/datadog/tracing/interceptor.go (hunk 2 above)
Comment (cretz) [reply contrib/datadog/tracing/interceptor.go:147]: I think `ddtrace.SpanFromContext(nil)` will do it

Comment (cdimitroulas) [reply contrib/datadog/tracing/interceptor.go]: Thank you for the suggestion, I have updated it to use this approach now

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: contrib/datadog/tracing/interceptor.go (hunk 2 above)
Comment (Quinn-With-Two-Ns) [reply contrib/datadog/tracing/interceptor.go]: Great thank you!

**Pull request merged**